### PR TITLE
Allows mappers to set their own title screen for each map

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Instead of having a picture of boxstation for every map.

Wishful thinking maybe,  but if we ever get more art, we can use it.

### CODE REVIEW

**Comment by ChangelingRain** [comment]
>as if anyone remains who can produce start screens like the one we currently have

**Comment by KorPhaeron** [comment]
@ChangelingRain 

The guy who made that one still plays, or at least hangs around the forums

**Comment by xxalpha** [comment]
who was it

**Comment by WJohn** [comment]
Invisty made our current title screen like 3 years ago, back on the first forums.

**Comment by xxalpha** [comment]
it had to be more than 3 years ago

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -299,6 +299,12 @@
 	icon = 'icons/misc/fullscreen.dmi'
 	icon_state = "title"
 	layer = FLY_LAYER
+	var/titlescreen = TITLESCREEN
```
Comment (neersighted) [inline code/game/turfs/turf.dm]: Why is this var on the base turf?

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -299,6 +299,12 @@
...
+
+/turf/indestructible/splashscreen/New()
+	..()
+	if(titlescreen)
+		icon_state = TITLESCREEN
```
Comment (neersighted) [inline code/game/turfs/turf.dm]: Why have titlescreen = TITLESCREEN?
Maybe `icon_state = titlescreen`.

**Comment by Iamgoofball** [comment]
Wait, Invisty made the title screen shit?

Did he make the animated shit too?

File: _maps/tgstation2.dm [hunk 3]
```diff
@@ -15,6 +15,8 @@ z7 = empty space
 
 #if !defined(MAP_FILE)
 
+		#define TITLESCREEN "title" //Add an image in misc/fullscreen.dmi, and set this define to the icon_state, to set a custom titlescreen for your map
```
Comment (tkdrg) [inline _maps/tgstation2.dm]: you gotta add this to efficiency as well

File: code/game/turfs/turf.dm (hunk 2 above)
Comment (tkdrg) [inline code/game/turfs/turf.dm]: what neer said

File: _maps/tgstation2.dm (hunk 3 above)
Comment (KorPhaeron) [inline _maps/tgstation2.dm]: Is it actually required? There is a check in New() to make sure it is defined before it attempts to set it

Comment (tkdrg) [inline _maps/tgstation2.dm]: its not required but you added it to every single one so i assume you want to

**Comment by freerealestate** [comment]
always been curious how our animations/title graphics were made

**Comment by KorPhaeron** [comment]
@tkdrg 

Okay it is updated

**Pull request merged**